src/library/trackset/crate/cratefeature.cpp

```diff
@@ -538,6 +546,33 @@ void CrateFeature::slotToggleCrateLock() {
     }
 }
 
+void CrateFeature::slotToggleCrateArchived() {
+    Crate crate;
+    if (readLastRightClickedCrate(&crate)) {
+        bool archiving = !crate.isArchived();
+        // If we are archiving the crate, select the crate below rather than following
+        // the crate into the archive subtree.
+        int reselectRow = 0;
+        if (archiving) {
+            const auto index = indexFromCrateId(crate.getId());
+            // Constrain reselection between first and last valid crate positions.
+            reselectRow = std::min(std::max(index.row(), 0), m_pSidebarModel->rowCount() - 3);
+        }
+        crate.setArchived(archiving);
+        if (!m_pTrackCollection->updateCrate(crate)) {
+            qDebug() << "Failed to toggle archive status of crate" << crate;
```

Comment:
What is the scenario for this? 
Can you give more details to also desringush this message from the one below?